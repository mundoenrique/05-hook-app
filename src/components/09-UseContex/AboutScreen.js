import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export default function AboutScreen() {
	const { user, setUser } = useContext(UserContext);
	return (
		<>
			<h3>AboutScreen</h3>
			<pre>{JSON.stringify(user, null, 2)}</pre>
			<button className="btn btn-outline-info" onClick={() => setUser({})}>
				Logout
			</button>
		</>
	);
}
