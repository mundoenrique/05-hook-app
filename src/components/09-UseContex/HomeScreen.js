import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export default function HomeScreen() {
	const { user } = useContext(UserContext);

	return (
		<>
			<h3>HomeScreen</h3>
			<pre>{JSON.stringify(user, null, 2)}</pre>
		</>
	);
}
