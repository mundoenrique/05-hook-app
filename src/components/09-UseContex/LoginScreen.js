import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export default function LoginScreen() {
	const { setUser } = useContext(UserContext);

	return (
		<>
			<h3>LoginScreen</h3>
			<button
				className="btn btn-primary"
				onClick={() =>
					setUser({
						id: 1234,
						name: 'Enrique',
						email: 'mundoenrique@gmail.com',
					})
				}
			>
				Login
			</button>
		</>
	);
}
