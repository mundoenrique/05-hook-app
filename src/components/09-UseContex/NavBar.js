import { NavLink } from 'react-router-dom';

export default function NavBar() {
	return (
		<>
			<nav className="navbar navbar-expand-lg navbar-dark bg-dark">
				<div className="container-fluid">
					<NavLink to="#" className="navbar-brand">
						UseContext
					</NavLink>
					<div className="collapse navbar-collapse" id="navbarNavAltMarkup">
						<div className="navbar-nav">
							<NavLink
								to="home"
								className="nav-link active"
								aria-current="page"
							>
								Home
							</NavLink>
							<NavLink to="about" className="nav-link active">
								About
							</NavLink>
							<NavLink to="login" className="nav-link active">
								Login
							</NavLink>
						</div>
					</div>
				</div>
			</nav>
		</>
	);
}
