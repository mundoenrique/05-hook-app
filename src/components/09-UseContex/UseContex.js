import { Outlet } from 'react-router-dom';

export default function UseContex() {
	return (
		<>
			<h2>UseContex</h2>
			<Outlet />
		</>
	);
}
