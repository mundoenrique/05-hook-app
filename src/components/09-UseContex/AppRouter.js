import { Routes, Route } from 'react-router-dom';

import AboutScreen from './AboutScreen';
import HomeScreen from './HomeScreen';
import LoginScreen from './LoginScreen';
import NavBar from './NavBar';

export default function AppRouter() {
	return (
		<div>
			<NavBar />
			<div className="container">
				<Routes>
					<Route path="home" element={<HomeScreen />} />
					<Route path="about" element={<AboutScreen />} />
					<Route path="login" element={<LoginScreen />} />
				</Routes>
			</div>
		</div>
	);
}
