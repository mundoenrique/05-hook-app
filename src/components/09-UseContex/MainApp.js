import { useState } from 'react';
import AppRouter from './AppRouter';
import { UserContext } from './UserContext';

export default function MainApp() {
	const [user, setUser] = useState({});

	return (
		<UserContext.Provider
			value={{
				user,
				setUser,
			}}
		>
			<AppRouter />
		</UserContext.Provider>
	);
}
