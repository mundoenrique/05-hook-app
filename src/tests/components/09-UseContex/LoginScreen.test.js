import { mount } from 'enzyme';
import LoginScreen from '../../../components/09-UseContex/LoginScreen';
import { UserContext } from '../../../components/09-UseContex/UserContext';

describe('Pruebas sobre <LoginScreen />', () => {
	const setUser = jest.fn();

	const wrapper = mount(
		<UserContext.Provider value={{ setUser }}>
			<LoginScreen />
		</UserContext.Provider>
	);

	test('Debe renderizar <LoginScreen />', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe ejecutar setUser con los argumentos esperados', () => {
		const user = {
			id: 1234,
			name: 'Enrique',
			email: 'mundoenrique@gmail.com',
		};

		wrapper.find('button').prop('onClick')(() => setUser(user));

		expect(setUser).toHaveBeenCalled();
		expect(setUser).toHaveBeenCalledWith(user);
	});
});
