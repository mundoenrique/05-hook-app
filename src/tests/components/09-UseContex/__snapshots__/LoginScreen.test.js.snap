// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <LoginScreen /> Debe renderizar <LoginScreen /> 1`] = `
Array [
  <h3>
    LoginScreen
  </h3>,
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>,
]
`;
